catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -710,6 +731,130 @@ public void setAttribute(Attribute attribute) {
         }
     }
 
+    private boolean verifyTypesMatch(Attribute attribute) {
+        return verifyTypesMatch(attribute.getName(),
+                attribute.getValues()
+                        .size() > 1 ? (Serializable) attribute.getValues() : attribute.getValue());
+    }
+
+    private boolean verifyTypesMatch(String name, Serializable value) {
+
+        AttributeDescriptor descriptor = this.type.getAttributeDescriptor(name);
+        if (descriptor != null && value != null && !StringUtils.isEmpty(value.toString())) {
+            AttributeType type = descriptor.getType();
+            if (type != null) {
+                Class binding = type.getBinding();
+                if (binding != null && value.getClass() != binding) {
+                    if (value instanceof List) {
+                        List valueList = (List) value;
+                        if (!valueList.isEmpty() && valueList.get(0) != null && valueList.get(0)
+                                .getClass() != binding) {
+                            LOGGER.warn(String.format(
+                                    "Attempted to assign %s to %s. The correct type for %s is %s.",
+                                    valueList.get(0).getClass(),
+                                    name,
+                                    name,
+                                    binding));
+                            return false;
+                        }
+                    } else {
+                        LOGGER.warn(String.format(
+                                "Attempted to assign %s to %s. The correct type for %s is %s.",
+                                value.getClass(),
+                                name,
+                                name,
+                                binding));
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
+    }
+
+    private AttributeImpl convertType(Attribute attribute) {
+        return convertType(attribute.getName(),
+                attribute.getValues()
+                        .size() > 1 ? (Serializable) attribute.getValues() : attribute.getValue());
+    }
+
+    private AttributeImpl convertType(String name, Serializable value) {
+        AttributeType.AttributeFormat attributeFormat = this.type.getAttributeDescriptor(name)
+                .getType()
+                .getAttributeFormat();
+
+        return new AttributeImpl(name, convertValue(attributeFormat, value));
+    }
+
+    private Serializable convertValue(AttributeType.AttributeFormat attributeFormat,
+            Serializable value) {
+
+        Serializable convertedValue = null;
+        SimpleDateFormat metacardDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```

Comment:
Can we make these `SimpleDateFormat` member variables so we're not creating new objects every time we convert types?